xow-get-firmware.sh.in

```diff
@@ -0,0 +1,97 @@
+#!/bin/bash
+# vim: et sts=4 sw=4
+
+set -euo pipefail
+
+readonly firmware=#DATADIR#/FW_ACC_00U.bin
+force=0
+
+usage() {
+    echo
+    echo "Usage: $(basename $0) [OPTIONS]"
+    echo
+    echo "Downloads the firmware required by XOW. Saved as $firmware"
+    echo
+    echo "OPTIONS"
+    echo
+    echo "  --force"
+    echo "        Overwrite existing file"
+    echo
+    echo "  --help"
+    echo "        This help screen"
+    echo
+}
+
+parse_args() {
+    while [[ $# -gt 0 ]]; do
+        case "$1" in
+            -h | --help ) usage; exit 0;;
+            -f | --force ) shift; force=1;;
+            * ) usage; exit 1;;
+        esac
+    done
+}
+
+check_firmware() {
+    ! [[ -e $firmware ]] && return
+
+    if (( force == 0 )); then
+        echo "Firmware file $firmware already exists. Exiting."
+        exit 0
+    fi
+
+    echo "Firmware file $firmware exists. Removing file."
+    pkexec rm $firmware
+}
+
+disclaimer() {
+    local -r url="https://www.microsoft.com/en-us/legal/intellectualproperty/copyright"
+    local -r title="Microsoft Terms of Use"
+    local -r text="Firmware is distributed under the Microsoft Terms of Use.\n
+                    Do you accept the terms of the agreement?"
+
+    xdg-open "$url"
```

Comment:
relying on user input is bad for automated systems like LPF

Reply:
Not sure what LPF is, how do such systems deal with EULA and similar legalise?
I'm hesitant to remove the prompt without a good set of examples (projects, documentation) clearly showing why it's safe.

Reply:
[lpf](https://github.com/leamas/lpf) `It works by downloading sources, possibly requiring a user to accept license terms and then building and installing a rpm package locally.`

I already use lpf to create a xow package for Fedora
![](https://media.discordapp.net/attachments/734401204330692629/862385954747842620/unknown.png?width=960&height=505)

If this PR goes through the script would be great to use with lpf to automatically download the firmware but there is no need for a prompt like this and might even fail because its being build as a different user

Reply:
Few things come to mind - seems like lpf is an Fedora only, sort of a workaround for packages like these.

Off the top of my head the current approach/PR allows distributions, to do a proper package since the licensing/ToS question is deferred to the user and only when needed.

Reply:
yes but it proves the existance of such automated solution that would be hindered by this script

Reply:
The new `--accept-terms` can be used.